tf_agents/agents/ppo/ppo_agent.py

```diff
@@ -118,6 +118,7 @@ def __init__(self,
                normalize_rewards=True,
                reward_norm_clipping=10.0,
                normalize_observations=True,
+               train_only_on_full_episodes=True,
```

Comment:
@oars should we make the default false here?

Reply:
I would also like to know whether there are any reasons behind this `valid_mask`?
I didn't see this implemented in any common RL libraries such as OpenAI Baselines or Ray RLlib.

Reply:
The mask is needed because of the way data is added to the RB. In general observers are not filtering transitions between `step_type.LAST` and `step_type.FIRST`. I also don't think all agents are handling this correctly, in general we need to fix this.

@ebrevdo defaulting to False would definetly be the safest option to make sure behavior doesn't change for users.

@sguada Know if any other parts of PPO rely on the experience being full episodes?

Reply:
I think what we need is to bootstrap the end of episodes using the value function if it is not an end of episode.

But we still need to use the mask to avoid invalid transitions between `LAST` and `FIRST`.

Rename to `train_only_on_partial_episodes=True,`

Reply:
New commits solve the described issues.